velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -503,5 +503,25 @@ TEST_F(DateTimeFunctionsTest, nextDay) {
   EXPECT_EQ(nextDay("2015-07-23", ""), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, hour) {
+  const auto hour = [&](std::optional<Timestamp> date) {
+    return evaluateOnce<int32_t>("hour(c0)", date);
+  };
+
+  EXPECT_EQ(std::nullopt, hour(std::nullopt));
+  EXPECT_EQ(0, hour(util::fromTimestampString("2024-01-08 00:23:00.001")));
```

Comment:
Maybe, better to move `fromTimestampString` into the lambda to make test code concise.
It would be better to add more various test cases, e.g., 1970-01-01 & the time before it.

Reply:
Sure, thanks for advising. Modified